### PR TITLE
Support comments in recipes

### LINKED ISSUE #15385: Recipes do not support comments
The recipe reader doesn't support comments in Json (it's a mode that isn't turned on).

Which is annoying when you have a lot of recipes, and need to make notes in them, as to why you did something.

Not sure if the System.Text pr has gone in yet, might change with that, as it's a Newtonsoft config that needs to be set.

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #15385

/cc @deanmarcussen

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Abstractions/Json/JOptions.cs [hunk 1]
```diff
@@ -62,8 +62,8 @@ static JOptions()
 
         Document = new JsonDocumentOptions
         {
-            CommentHandling = Default.ReadCommentHandling,
-            AllowTrailingCommas = Default.AllowTrailingCommas,
+            CommentHandling = JsonCommentHandling.Skip,
+            AllowTrailingCommas = true,
```
Comment (MikeAlhayek) [inline src/OrchardCore/OrchardCore.Abstractions/Json/JOptions.cs:66]: There is no value change here. I just did not like how we are depending on the Default options.

File: src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/ImportController.cs [hunk 2]
```diff
@@ -129,7 +130,7 @@ public async Task<IActionResult> Json(ImportJsonViewModel model)
                 return Forbid();
             }
 
-            if (!model.Json.IsJson())
+            if (!model.Json.IsJson(null, JOptions.Document))
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/ImportController.cs:133]: Is the rest of the deployment logic using the same options to parse the document?

File: src/OrchardCore/OrchardCore.Abstractions/Shell/Configuration/Internal/JsonConfigurationParser.cs [hunk 3]
```diff
@@ -54,15 +54,9 @@ public sealed class JsonConfigurationParser
...
-        var jsonDocumentOptions = new JsonDocumentOptions
-        {
-            CommentHandling = JsonCommentHandling.Skip,
-            AllowTrailingCommas = true,
-        };
-
         try
         {
-            using (var doc = await JsonDocument.ParseAsync(input, jsonDocumentOptions))
+            using (var doc = await JsonDocument.ParseAsync(input, JOptions.Document))
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Abstractions/Shell/Configuration/Internal/JsonConfigurationParser.cs:59]:
```suggestion
            // Use JOptions.Document to allow comments and trailing commas
            using (var doc = await JsonDocument.ParseAsync(input, JOptions.Document))
```

File: src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/ImportController.cs (hunk 2 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/ImportController.cs:133]: yes 
https://github.com/OrchardCMS/OrchardCore/blob/3bd5ae8a7bbf1e9b2a924e6a15d4905494a49b2b/src/OrchardCore/OrchardCore.Recipes.Core/Services/RecipeExecutor.cs#L57

File: src/OrchardCore/OrchardCore.Mvc.Core/Utilities/StringExtensions.cs [hunk 4]
```diff
@@ -501,11 +502,11 @@ public static string ToPascalCase(this string attribute, char upperAfterDelimite
         /// <summary>
         /// Tests if a string is valid json.
         /// </summary>
-        public static bool IsJson(this string json)
+        public static bool IsJson(this string json, JsonNodeOptions? jsonNodeOptions = null, JsonDocumentOptions jsonDocumentOptions = default)
```
[the remark this answers is not in the document]
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.Mvc.Core/Utilities/StringExtensions.cs:505]: should I just remove the JsonNodeOption or move it as the last parameter?

[the remark this answers is not in the document]
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.Mvc.Core/Utilities/StringExtensions.cs]: I removed it for now. If needed, we can always add a new parameter with null ad default value.

**Review by sebastienros — APPROVED** [review]

**Pull request merged**